src/nvim/ugrid.c

```diff
@@ -72,8 +73,10 @@ void ugrid_scroll(UGrid *grid, int top, int bot, int left, int right, int count)
   for (i = start; i != stop; i += step) {
     UCell *target_row = grid->cells[i] + left;
     UCell *source_row = grid->cells[i + count] + left;
+    size_t bytes;
```

Comment:
columns, not bytes. Though this doesn't look very readable to me, I would prefer something like `assert(right >= left)` and then write the calculation as is.

Reply:
```
right = -3
left = -1
```

`assert(right >= left)` the condition holds and we still get  negative value here `right - left + 1`  that will be cast to unsigned. I know that this situation is unlikely but I think being more defensive with memory operations will not be misplaced here.

We could check that both operands are positive to avoid this  `assert(right >= left && left >= 0)` 
Or maybe  `assert(right >= left && right >= 0)`  if  `left` could be negative

Reply:
probably `assert(right >= abs(left) && right >= 0)` is more correct

Reply:
none of these should ever be negative. Just check `assert(right >= left && left >= 0)` if we want to be paranoid.

Reply:
being paranoid is a duty of C programmer (c)

Reply:
fixed

Reply:
Indeed :) Though also ASAN does a lot of the paranoia for us.